src/handler/mod.rs

```diff
@@ -360,6 +372,7 @@ impl Handler {
         );
         self.target_ids.push(target.target_id().clone());
         self.targets.insert(target.target_id().clone(), target);
+        self.event_listeners.start_send(event);
```

Comment:
This is special for a single event. Would be much more beneficial if we could figure out a more generic way to handle all events